### ISSUE TITLE
ContentDialog's Title does not have HeadingLevel automation property set

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/microsoft-ui-xaml#10946]
### Describe the bug

The title for a content dialog should have the [AutomationProperties.HeadingLevel](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.automationproperties.headinglevel?view=winrt-26100) set to something (probably Level1 for simplicity).

### Why is this important?

Users who rely on screen reader will feel difficulty in analyzing the structure of the page, if visually appearing heading is not defined as heading programmatically.

WCAG Reference link:
[1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships)

### Steps to reproduce the bug

1. Open WinUI 2 Gallery > Dialogs and Flyouts > [ContentDialog](winui2gallery://item/ContentDialog)
2. Click "Show dialog"
3. Use accessibility insights to check the heading level of the title of the ContentDialog

### Actual behavior

no heading level set

### Expected behavior

heading level should be set

### Screenshots

<img width="1240" height="501" alt="Image" src="https://github.com/user-attachments/assets/4eedc1ac-6f11-4132-a2ea-fafd3ae566c2" />

### NuGet package version

None

### Additional context

Originally filed on Windows Terminal: https://github.com/microsoft/terminal/issues/19686
